packages/gensx-openai/src/structured-output.tsx

```diff
@@ -49,29 +49,37 @@ export const structuredOutputImpl = async <T,>(
       }
 
       // Make initial completion
-      const completion = await gsx.execute<ChatCompletionOutput>(
+      let completion = await gsx.execute<ChatCompletionOutput>(
         <OpenAIChatCompletion
           {...rest}
-          messages={messages}
+          messages={currentMessages}
           tools={tools?.map((t) => t.definition)}
           response_format={zodResponseFormat(outputSchema, "output_schema")}
         />,
       );
 
       const toolCalls = completion.choices[0].message.tool_calls;
       // If we have tool calls, execute them and make another completion
-      if (toolCalls?.length && tools) {
-        const toolResult = await gsx.execute<ChatCompletionOutput>(
-          <ToolExecutor
-            tools={tools}
-            toolCalls={toolCalls}
-            messages={[...messages, completion.choices[0].message]}
-            model={rest.model}
+      while (toolCalls?.length && tools) {
```

Comment:
Is this supposed to be a while loop? I don't see where `toolCalls` is modified.

Reply:
yeah, it needs to be a while loop in case the model does multiple rounds of tool calls...but you make a good point. This will always only ever run once because there's a return statement in the loop.

Reply:
okay fixed and confirmed it works by temproarily disabling parallel tool calls: https://app.gensx.com/gensx/executions/01JMX7GP1J46Z8EZE2BSBZ69Q0?workflowName=ToolsWithStructuredOutputWorkflow